src/Components/test/testassets/TestServer/Pages/ComponentWithParameters.cshtml

```diff
@@ -0,0 +1,37 @@
+@page
+
+<component type="typeof(ComponentWithParameters)"
+           render-mode="ServerPrerendered"
+           parameter-Param1="ComponentWithParameters.TestModelValues"
+           parameter-Param2="ComponentWithParameters.DerivedModelValue"
```

Comment:
This is kinda an interesting case. It's interesting to think about because the same is not true with `Param1`. If you put an instance of `DerivedModel` in that list, it wouldn't roundtrip.

I think is is probably the right thing to do but I'm sure it's going to come up. This use case for serialization is really different from our usage elsewhere. We'll have to make sure to write good docs for this - the main thing is that it's not really possible for people to marshal some kinds of data (like polymorphic collections) without doing a lot of S.T.J extensibility.

Reply:
Yup. I wanted to at least capture some of the ordinary cases. The experience is generally going to be bad with polymorphism. `Dictionary<string, object>` for instance are going to round trip as `JsonElement` instances. I'll make sure to specifically say that the implementation uses S.T.J to marshal the values so we recommend simple values that are easy to round trip, don't exceed the size limits, etc